GolemLib/IFactory.cs

```diff
@@ -4,5 +4,11 @@ namespace GolemLib;
 
 public interface IFactory
 {
-    public Task<IGolem> Create(string modulesDir, ILoggerFactory? loggerFactory, bool mainnet = true);
+    /// <summary>Creates Golem object</summary>
+    /// <param name="mainnet">Enables usage of mainnet payment network. By default `true`</param>
+    /// <param name="dataDir">Path where `provider` and `yagna` data directories will be created.
+    /// If not specified it will default to `golem-data` dir inside of Golem package directory.
+    /// </param>
+    /// <returns>Golem object</returns>
+    public Task<IGolem> Create(string modulesDir, ILoggerFactory? loggerFactory = null, bool mainnet = true, string? dataDir = null);
```

Comment:
I converted `loggerFactory` into optional param (by `= null`). It should not break anything.